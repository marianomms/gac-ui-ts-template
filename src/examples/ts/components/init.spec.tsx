import React from 'react';
import * as Immutable from 'immutable';
import shallowWithStore from '__spec_helper__/shallow-helper';

import Init from './init';

describe('<Init />', () => {
  const state = {
    exampleReducer: {
      exampleText: 'initial value in the state'
    }
  };
  const store = Immutable.fromJS(state);

  it('renders the component', () => {
    expect.assertions(1);
    const wrapper = shallowWithStore(<Init myProp1='baz' />, store);
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.find('Init').dive()).toMatchSnapshot();
  });
});