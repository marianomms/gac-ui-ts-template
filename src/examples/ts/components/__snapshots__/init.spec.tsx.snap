// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Init /> renders the component 1`] = `
<AppWrapper
  className="add-transactions-page"
>
  <Heading
    divider={true}
    helpLink="foo"
    separator={false}
    subheader="subheader, I am"
    title="I'm the title"
  />
  <p>
    baz
  </p>
  <p
    className="red"
  >
    initial value in the state
  </p>
  <p>
    <Button
      as="secondary"
      destructive={false}
      disabled={false}
      iconPosition="before"
      onClick={[Function]}
      size="medium"
      styleOverride={
        Object {
          "icon": Object {},
          "root": Object {},
        }
      }
      subtext=""
    >
      Change text
    </Button>
  </p>
</AppWrapper>
`;
